### PR TITLE
[docs] Update Installation-Guide.rst for macOS

### PR DESCRIPTION
**Pull request by thierrygosselin**
*  "brew-less" detailed instructions

### CODE REVIEW

File: docs/Installation-Guide.rst [hunk 1]
```diff
@@ -84,26 +84,71 @@ LightGBM uses **CMake** to build. Run the following commands:
...
+   cd Downloads
+   wget http://prdownloads.sourceforge.net/hpc/gcc-7.1-bin.tar.gz
+   # This will install the compiler in `/usr/local/bin`
+   sudo tar -zxvf gcc-7.1-bin.tar.gz -C/
+   # to remove the package from your Downloads directory
+   sudo rm gcc-7.1-bin.tar.gz
+   # to check the installation
+   gcc -v
+
+Install *`CMake <https://cmake.org/overview/>`_*
```
Comment (Laurae2) [inline docs/Installation-Guide.rst]: Fix how the link is shown

Comment (Laurae2) [inline docs/Installation-Guide.rst]: Why not `brew install cmake`?

File: docs/Installation-Guide.rst [hunk 2]
```diff
@@ -84,26 +84,71 @@ LightGBM uses **CMake** to build. Run the following commands:
...
+
+Then, run the following commands to install LightGBM:
+
+.. code::
+
+  cd Downloads
+  git clone --recursive https://github.com/Microsoft/LightGBM
+  cd LightGBM; mkdir build ; cd build
+  export CC=/usr/local/bin/gcc CXX=/usr/local/bin/g++ CXX1X=/usr/local/bin/g++ # with GCC
+  cmake ../CMakeLists.txt
```
Comment (Laurae2) [inline docs/Installation-Guide.rst]: do we need `../CMakeLists.txt` specifically? (`cmake ..` should be enough)

File: docs/Installation-Guide.rst [hunk 3]
```diff
@@ -179,26 +224,25 @@ Then run the following commands:
...
 
-  git clone --recursive https://github.com/Microsoft/LightGBM ; cd LightGBM
-  export CXX=g++-7 CC=gcc-7
-  mkdir build ; cd build
-  cmake -DUSE_MPI=ON ..
-  make -j4
+   git clone --recursive https://github.com/Microsoft/LightGBM ; cd LightGBM
+   export CC=/usr/local/bin/gcc CXX=/usr/local/bin/g++ CXX1X=/usr/local/bin/g++
+   mkdir build ; cd build
+   cmake -DUSE_MPI=ON ../CMakeLists.txt
```
Comment (Laurae2) [inline docs/Installation-Guide.rst]: do we need `../CMakeLists.txt` specifically? (`cmake ..` should be enough)

File: docs/Installation-Guide.rst [hunk 4]
```diff
@@ -84,26 +84,71 @@ LightGBM uses **CMake** to build. Run the following commands:
 
 Also you may want to reed `gcc Tips <./gcc-Tips.rst>`__.
 
-OSX
-~~~
+macOS
```
Comment (Laurae2) [inline docs/Installation-Guide.rst]: OSX different vs macOS?

File: docs/Installation-Guide.rst [hunk 5]
```diff
@@ -179,26 +224,25 @@ Then run the following commands:
 
 **Note**: glibc >= 2.14 is required.
 
-OSX
-^^^
-
-Install **gcc** and **Open MPI** first:
+macOS
```
Comment (Laurae2) [inline docs/Installation-Guide.rst]: OSX different vs macOS?

File: docs/Installation-Guide.rst [hunk 6]
```diff
@@ -179,26 +224,25 @@ Then run the following commands:
...
+See instruction above to install **gcc** and **CMake**
 
 .. code::
 
-  brew install openmpi
-  brew install cmake
-  brew install gcc --without-multilib
+   brew install openmpi
+   brew install cmake
+   brew install gcc --without-multilib
```
Comment (Laurae2) [inline docs/Installation-Guide.rst]: Why do we need brew now and not before?

File: docs/Installation-Guide.rst (hunk 5 above)
Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: it's like saying Linux OS
OSX is a version
macOS in the OS

File: docs/Installation-Guide.rst (hunk 3 above)
Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: `cmake ..` doesn't work
cmake needs the path to the file on macOS

File: docs/Installation-Guide.rst (hunk 2 above)
Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: yes see below

File: docs/Installation-Guide.rst (hunk 1 above)
Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: I prefer to go directly with **cmake**
Personal experience:
In genomics, I've lost tremendous amount of time to debug when things were installed with *homebrew* or *macport*, now I stay far far away!

File: docs/Installation-Guide.rst (hunk 6 above)
Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: For open MPI there a bug right now with the lates macOS to install directly.
the standard :`./configure`, `make`; `sudo make install` no longer work.

I didn't install the MPI version, will wait for the bug fix.

File: docs/Installation-Guide.rst (hunk 5 above)
Comment (StrikerRUS) [reply docs/Installation-Guide.rst]: To be more precise:
>macOS (/ˌmækoʊˈɛs/; previously Mac OS X and later OS X) is the current series of Unix-based graphical operating systems developed and marketed by Apple Inc. ...

[wiki1](https://en.wikipedia.org/wiki/MacOS), [wiki2](https://en.wikipedia.org/wiki/OSX_(disambiguation))
So previously it was **OS X**, but now **macOS** is a right name.

File: docs/Installation-Guide.rst [hunk 7]
```diff
@@ -84,26 +84,71 @@ LightGBM uses **CMake** to build. Run the following commands:
...
+Then, run the following commands to install LightGBM:
+
+.. code::
+
+  cd Downloads
+  git clone --recursive https://github.com/Microsoft/LightGBM
+  cd LightGBM; mkdir build ; cd build
+  export CC=/usr/local/bin/gcc CXX=/usr/local/bin/g++ CXX1X=/usr/local/bin/g++ # with GCC
+  cmake ../CMakeLists.txt
+  cd ..
```
Comment (StrikerRUS) [inline docs/Installation-Guide.rst]: Why do we need this (and don't need in the MPI case?)?

File: docs/Installation-Guide.rst [hunk 8]
```diff
@@ -179,26 +224,25 @@ Then run the following commands:
 
 **Note**: glibc >= 2.14 is required.
 
-OSX
-^^^
-
-Install **gcc** and **Open MPI** first:
+macOS
+^^^^^
+See instruction above to install **gcc** and **CMake**
```
Comment (StrikerRUS) [inline docs/Installation-Guide.rst]: Please add a blank line after the heading and a link: ```See instruction `above <#macos>`__```.

File: docs/Installation-Guide.rst (hunk 6 above)
Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: Here is the issue for the openMPI problem referred to
https://github.com/open-mpi/ompi/issues/4373#issuecomment-338695218

And the fix is to use clang to compile... I have tried and it's working

File: docs/Installation-Guide.rst (hunk 7 above)
Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: Did not test with MPI

Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: I was not able to have a successful build with MPI as stated above, but yes it's needed too.

Comment (StrikerRUS) [reply docs/Installation-Guide.rst]: We currently have the following working piece of code:
https://github.com/Microsoft/LightGBM/blob/66c81af2a62fa367063c3d180e58087ab3eeb4ae/.travis/test.sh#L97-L106

And there is no `cd ..` here.

Comment (thierrygosselin) [reply docs/Installation-Guide.rst]: We are not talking about the same thing.
Have you at least tried it ?

Comment (StrikerRUS) [reply docs/Installation-Guide.rst]: No, I didn't try your code.

Comment (wxchan) [reply docs/Installation-Guide.rst]: @Laurae2 and @StrikerRUS is right. You don't need cmake ../CMakeLists.txt, you are installing all the things into LightGBM folder, not build folder. You don't even need mkdir folder in this case.

Our current installation steps work both on my macOS and our travis test machine.

**Pull request closed without merging**